### ISSUE TITLE
Windows users

### ISSUE THREAD

**Issue by almeidaz** [squizlabs/sublime-PHP_CodeSniffer#3]
Hi, This is a looking good one plugin ! I'm a windows user and i wanted to know if it's compatible with windows users.
i ve download from the repositery and i ve rename the folder from "sublime-PHP_CodeSniffer-master" to "PHP_CodeSniffer" then put him in \Roaming\Sublim text 3\Packages\
i have edited my settings files with this value :

edit after post : "we didn't see double\ in path on the post of this message but they really exist !

// Path to PHP.
"php_path": "C:\Program Files (x86)\EasyPHP\binaries\php\php_runningversion\php.exe",

// Path to the PHPCS script.
"phpcs_path": "C:\php\pear\phpcs.bat",

// Path to the PHPCBF script.
"phpcbf_path": "C:\php\pear\phpcbf.bat",

I've got this error in sublim text console :

@echo off
REM PHP_CodeSniffer tokenizes PHP code and detects violations of a
REM defined set of coding standards.
REM
REM PHP version 5
REM
REM @category PHP
REM @package PHP_CodeSniffer
REM @author Greg Sherwood gsherwood@squiz.net
REM @author Marc McIntyre mmcintyre@squiz.net
REM @copyright 2006-2014 Squiz Pty Ltd (ABN 77 084 670 600)
REM @license https://github.com/squizlabs/PHP_CodeSniffer/blob/master/licence.txt BSD Licence
REM @link http://pear.php.net/package/PHP_CodeSniffer

if "%PHPBIN%" == "" set PHPBIN=C:\Program Files (x86)\EasyPHP\binaries\php\php_runningversion\php.exe
if not exist "%PHPBIN%" if "%PHP_PEAR_PHP_BIN%" neq "" goto USE_PEAR_PATH
GOTO RUN
:USE_PEAR_PATH
set PHPBIN=%PHP_PEAR_PHP_BIN%
:RUN
"%PHPBIN%" "C:\php\pear\phpcs" %*

Then i have create windows environment variable
PHPBIN with value `C:\Program Files (x86)\EasyPHP\binaries\php\php_runningversion\php.exe
And change :

// Path to PHP.
"php_path": "%PHPBIN%",

the error message in sublime text disappeared
but ! no error is detect in my php file :( seem to be not working.
Do you have a working PHP_CodeSniffer.sublime-settings --User Example ? Thx for help.

Sorry for posting here. I know it's not the best place but it's difficult to contact you. I will delete this post if you ask me. And sry for my english :)
Cheers. Thx in advance.

**Comment by sertand** [comment]
Sorry, I have never tried using the plugin on Windows so it will be very hard for me to help you. My first suggestion would be to try running PHPCS on its own to see if it is actually working without using the plugin.

**Comment by almeidaz** [comment]
Yes phpcs works in command line. But phpcbf doesnt recognize diff. I need to add  --no-patch. There is fc command on windows similar as diff. This is the problem i think.

**Comment by almeidaz** [comment]
I'm installing Cygwin to get those command. ! I will see.

**Comment by almeidaz** [comment]
Nothing change, plugin doesn't work for me.

**Comment by sertand** [comment]
I have just got my hands on Windows and installed the plugin. It worked with these settings:

``` javascript
{
    "phpcs_path": "C:\\PHP_CodeSniffer\\scripts\\phpcs",
    "phpcbf_path": "C:\\PHP_CodeSniffer\\scripts\\phpcbf"
}
```

I did not need to specify the php_path. 
Also I was using checked out copy of PHP_CodeSniffer instead of  the one in Pear.

**Comment by almeidaz** [comment]
Thx for reply. 
by checked out copy you mean this one https://github.com/squizlabs/PHP_CodeSniffer ? 
do you have cygwin installed too?
Because i didnt install all package from cygwin. that very huge.
i have installed  "diff" and "patch" command. So if you think about command i need to install, tell me.
thx you for giving me hope :)

**Comment by sertand** [comment]
Yes, just clone PHP_CodeSniffer. I did not install cygwin or anything else. This plugin does not require you to install diff or patch, this is handled by the plugin.

**Comment by almeidaz** [comment]
Voila ! It's working ! Thx for your help.
then now i use this version  https://github.com/squizlabs/PHP_CodeSniffer
installed at c:\php\PHP_CodeSniffer\ for me
Works great in ST3 , Just Perfect.


So i wanted to use phpcbf.bat and phpcbs.bat in Windows CMD too.
I had c:\php\PHP_CodeSniffer\scripts; in environment variable PATH 
I had to edit phpcs.bat at line 
::"%PHPBIN%" "@bin_dir@\phpcs" %*
"%PHPBIN%" "C:\php\PHP_CodeSniffer\scripts\phpcs" %*

same thing for phpcbf.bat at line 
:: "%PHPBIN%" "@bin_dir@\phpcbf" %*
"%PHPBIN%" "C:\php\PHP_CodeSniffer\scripts\phpcbf" %*

otherwhise phpcs in windows CMD say : @bin_dir@\phpcs says :
Could not open input file ; @bin_dir@\phpcs

at this point i can use phpcs and phpcb in windows CMD because i had "diff" and "patch" command via cygwin.

then
c:\2\phpcs test.php    Working !
c:\2phpcbf test.php --no-patch  Working !
c:\2\phpcbf test.php --report=diff    Working ! 

but 

c:\2\phpcbf test.php says:

Changing into directory C:\2
Processing test.php [PHP => 172 tokens in 25 lines]... DONE in 15ms (17 fixable violations
        => Fixing file: 0/17 violations remaining [made 3 passes]... ←[32mDONE←[0m in 60ms
Array
(
    [0] => patching file test.php
    [1] => Hunk #1 FAILED at 1 (different line endings).
    [2] => 1 out of 1 hunk FAILED -- saving rejects to file test.php.rej
)
Returned: 1
Time: 418ms; Memory: 2.75Mb

what's the problem in this case?
( that mean in the sublim plugin you use --no-patch) ?

also
i've try to use --ReportFile but with no success
c:\2\phpcs test.php --ReportFile=test.rpf
or c:\2\phpcs test.php --ReportFile=c:\2\test.rpf
or c:\2\phpcs test.php --report=diff --reportFile=test.rpf
and many other with no success :)
were you able to use ReportFile on windows?

when c:\2\phpcbf test.php fail  he created  test.php.orig and test.php.rej

the .rej look nice but the one with your plugin in sublime console after fixed file looks better.It is possible to save in file the "console log". i mean for example :

-if (!defined('_PS_VERSION_'))
-	exit;
+if (!defined('_PS_VERSION_')) {
+    exit;
+}
 
 class Papp extends Module
 {
-	/* @var boolean error */
+    /* @var boolean error */
     protected $_errors = false;
-	protected $allowedExts = array("mp3","wav","ogg");
+    protected $allowedExts = array("mp3","wav","ogg");
 
-	public function __construct()
-      	{
 
maybe by adding somethings in  
 // Additional arguments to pass to PHPCS/PHPCBF.
     "additional_args": [],

thx for reading and many thanks.

**Comment by almeidaz** [comment]
It's working ! Thx for your help.

i added 

```
 "additional_args": [
 "--report-diff=%HOMEPATH%\\myreport.rpt",
],
```

this makes a file with diff for me. Do you know how can i replace %HOMEPATH% by the current directory of the sniffed file and replace myreport by the sniffed filename?

**Comment by leotm** [comment]
You guys manage `php-cs-fixer.bat` the manual way too?
Since there's not pear install :(
